service/controller/app/resource/releasemigration/create.go

```diff
@@ -54,11 +53,29 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 
 	if cr.Status.AppVersion == "" {
 		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("app %#q is not installed yet", key.AppName(cr)))
+	}
+
+	v, err := semver.NewVersion(cr.Status.AppVersion)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if v.Major() < 1 {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("app %#q with appVersion %#q is using Helm 2. we don't need to trigger Helm 3 migration.", key.AppName(cr), cr.Status.AppVersion))
 		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 		return nil
 	}
 
-	v, err := semver.NewVersion(cr.Status.AppVersion)
+	deploy, err := cc.Clients.K8s.K8sClient().AppsV1().Deployments(key.Namespace(cr)).Get(cr.Name, metav1.GetOptions{})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// extract spec container image
+	image := deploy.Spec.Template.Spec.Containers[0].Image
+	tag := strings.Split(image, ":")[1]
```

Comment:
In newer releases we also have a label on the deployed `app.kubernetes.io/version: 0.13.0`.

But we might need to upgrade from a version that misses this label. So I think using the image is safest. 👍